File: Python/SingleThreaded/crdFiles.py
import random
import time
import os

def generateRandList(size):
  L = []
  for i in range(0, size):
    L.append(random.randint(0, 32767))
  return L

def writeFile(filename, contents):
  with open(filename, 'w') as file:
    file.write(contents)

def readFile(filename):
  with open(filename, 'r') as file:
    return file.read()
  
def createReadDelete(filename, contents):
  writeFile(filename, contents)
  readFile(filename)
  os.remove(filename)

if __name__ == "__main__":
  # Setup and Generate Random number lists
  RandList = generateRandList(16)

  # Perform selection sorts and measure time
  start = time.perf_counter()
  for i in range(0, 16):
    name = f"randomFile{i}.txt"
    createReadDelete(name, str(RandList[i]))
  stop = time.perf_counter()
  duration = stop - start

  # Print Results
  print(f'{duration:5f}')

File: C++/MultiThreaded/crdFiles.cpp
#include <chrono>
#include <cstdio>
#include <fstream>
#include <iostream>
#include <thread>
#include <vector>

using namespace std::chrono;

std::vector<int> generateRandList(int size) {
  std::vector L(size, 0);
  for (int i = 0; i < L.size(); i++) {
    L[i] = std::rand();
  }
  return L;
}

void writeFile(const std::string &filename, int contents) {
  std::ofstream outputFile(filename);
  outputFile << contents << std::endl;
  outputFile.close();
}

int readFile(const std::string &filename) {
  std::ifstream inputFile(filename);
  std::string text;
  std::getline(inputFile, text);
  inputFile.close(); // Close the file when done
  return std::stoi(text);
}

void createReadDelete(const std::string &filename, int contents) {
  writeFile(filename, contents);
  readFile(filename);
  std::remove(filename.c_str()); // Deletes File
}

int main() {
  // Setup and Generate Random number lists
  std::srand(std::time(0));
  std::vector<int> RandList = generateRandList(16);
  std::vector<std::thread> threads;

  // Create, Read, and Delete Files + Time
  auto start = high_resolution_clock::now();
  for (int i = 0; i < RandList.size(); i++) {
    std::string name = "randomFile" + std::to_string(i) + ".txt";
    threads.push_back(std::thread(createReadDelete, name, RandList[i]));
  }
  for (auto &th : threads)
    th.join();
  auto stop = high_resolution_clock::now();
  auto duration = duration_cast<microseconds>(stop - start);

  // Print Results
  std::cout << static_cast<float>(duration.count()) / 1000000 << std::endl;

  return 0;
}
